static/app/views/performance/traceDetails/newTraceDetailsTraceView.tsx

```diff
@@ -0,0 +1,498 @@
+import {
+  TraceDetailBody,
+  TraceViewContainer,
+  TraceViewHeaderContainer,
+} from 'sentry/views/performance/traceDetails/styles';
+import TransactionGroup from 'sentry/views/performance/traceDetails/transactionGroup';
+import {TraceInfo, TreeDepth} from 'sentry/views/performance/traceDetails/types';
+import {
+  getTraceInfo,
+  hasTraceData,
+  isRootTransaction,
+} from 'sentry/views/performance/traceDetails/utils';
+
+import LimitExceededMessage from './limitExceededMessage';
+import {TraceType} from './newTraceDetailsContent';
+import TraceNotFound from './traceNotFound';
+
+type AccType = {
+  lastIndex: number;
+  numberOfHiddenTransactionsAbove: number;
+  renderedChildren: React.ReactNode[];
+};
+
+type Props = Pick<RouteComponentProps<{}, {}>, 'location'> & {
+  meta: TraceMeta | null;
+  organization: Organization;
+  rootEvent: EventTransaction | undefined;
+  traceEventView: EventView;
+  traceSlug: string;
+  traceType: TraceType;
+  traces: TraceFullDetailed[];
+  filteredEventIds?: Set<string>;
+  handleLimitChange?: (newLimit: number) => void;
+  orphanErrors?: TraceError[];
+  traceInfo?: TraceInfo;
+};
+
+function TraceHiddenMessage({
+  isVisible,
+  numberOfHiddenTransactionsAbove,
+  numberOfHiddenErrorsAbove,
+}: {
+  isVisible: boolean;
+  numberOfHiddenErrorsAbove: number;
+  numberOfHiddenTransactionsAbove: number;
+}) {
+  if (
+    !isVisible ||
+    (numberOfHiddenTransactionsAbove < 1 && numberOfHiddenErrorsAbove < 1)
+  ) {
+    return null;
+  }
+
+  const numOfTransaction = <strong>{numberOfHiddenTransactionsAbove}</strong>;
+  const numOfErrors = <strong>{numberOfHiddenErrorsAbove}</strong>;
+
+  const hiddenTransactionsMessage =
+    numberOfHiddenTransactionsAbove < 1
+      ? ''
+      : numberOfHiddenTransactionsAbove === 1
+      ? tct('[numOfTransaction] hidden transaction', {
+          numOfTransaction,
+        })
+      : tct('[numOfTransaction] hidden transactions', {
+          numOfTransaction,
+        });
+
+  const hiddenErrorsMessage =
+    numberOfHiddenErrorsAbove < 1
+      ? ''
+      : numberOfHiddenErrorsAbove === 1
+      ? tct('[numOfErrors] hidden error', {
+          numOfErrors,
+        })
+      : tct('[numOfErrors] hidden errors', {
+          numOfErrors,
+        });
+
+  return (
+    <MessageRow>
+      <span key="trace-info-message">
+        {hiddenTransactionsMessage}
+        {hiddenErrorsMessage && hiddenTransactionsMessage && ', '}
+        {hiddenErrorsMessage}
+      </span>
+    </MessageRow>
+  );
+}
+
+function isRowVisible(
+  row: TraceFullDetailed | TraceError,
+  filteredEventIds?: Set<string>
+): boolean {
+  return filteredEventIds ? filteredEventIds.has(row.event_id) : true;
+}
+
+function generateBounds(traceInfo: TraceInfo) {
+  return boundsGenerator({
+    traceStartTimestamp: traceInfo.startTimestamp,
+    traceEndTimestamp: traceInfo.endTimestamp,
+    viewStart: 0,
+    viewEnd: 1,
+  });
+}
+
+export default function NewTraceView({
```

Comment:
Added a new file for the traceview,  to isolate changes from having effect on the current traceview. No major changes right now just just prop changes and renders the new TraceViewHeader.